sdks/python/apache_beam/pipeline_test.py

```diff
@@ -318,8 +317,13 @@ def get_replacement_transform(self, ptransform):
           return TripleParDo()
         raise ValueError('Unsupported type of transform: %r', ptransform)
 
-    # Using following private variable for testing.
-    DirectRunner._PTRANSFORM_OVERRIDES.append(MyParDoOverride())
+    # Monkey patching a private variable for testing.
+    def get_overrides():
+      return [MyParDoOverride()]
+
+    from apache_beam.runners.direct import direct_runner
+    direct_runner._get_transform_overrides = get_overrides
```

Comment:
<!--new_thread; commit:7549b5c2ebe2ae47af9066eaf97364a27e828ab5; resolved:0-->
This sort of monkey-patching may not be safe--subsequent tests in the same process will get your modified version of `_get_transform_overrides`.  Consider using a `@mock.patch('apache_beam.runners.direct.direct_runner._get_transform_overrides')` annotation on this method (that machinery will restore the original value after this test case finishes).

Actually, alternatively, you can create the `DirectRunner` object and directly patch `runner._ptransform_overrides` instead, which might be cleaner.